common/djangoapps/third_party_auth/tests/specs/base.py

```diff
@@ -596,10 +612,14 @@ def test_full_pipeline_succeeds_for_unlinking_account(self):
         actions.do_complete(request.backend, social_views._do_login,  # pylint: disable=protected-access
                             request=request)
 
-        with self._patch_edxmako_current_request(strategy.request):
-            login_user(strategy.request)
-            actions.do_complete(request.backend, social_views._do_login, user=user,  # pylint: disable=protected-access
-                                request=request)
+        login_post_request = self._get_login_post_request(strategy)
+        with self._patch_edxmako_current_request(login_post_request):
+            login_user(login_post_request)
+            actions.do_complete(login_post_request.backend, social_views._do_login, user=user,  # pylint: disable=protected-access, no-member
+                                request=login_post_request)
+
+        # The logged in user from the earlier login_post_request is required
```

Comment:
Nit: consider rephrasing this comment to explain why the user needs to be stored in the `request` object.  Otherwise, it requires the reader to do extra effort to understand why it "is required".  Perhaps something like:

`Now that the user is set on the login_post_request object, copy the user back onto the original request object.`

Reply:
Alternatively, why keep separate `request` objects around?  Can you override the original `request` object with the new one generated for POSTs?

Reply:
@nasthagiri: I don’t think I can change the existing request, but I think I have a more clear vision for a minor refactor to respond to your comments. No need for you to update.